lib/ansible/module_utils/facts/hardware/linux.py

```diff
@@ -460,6 +463,13 @@ def _mtab_entries(self):
             mtab_entries.append(fields)
         return mtab_entries
 
+    def _replace_octal_escapes(self, value):
+        def _replace_escape(match):
```

Comment:
After some thought, I would like to see this is as a private or static method. Nested functions cannot be tested independently and are a bit harder to understand when reading since they are less common.

Reply:
The function isn't useful independently. It takes a regex match object, which only exists after a call to the `re.sub()` function. I could convert it to a function that just takes a simple value, but that would require another intermediate function (or a lambda) to pull the bare value from the regex match object.

Reply:
Would this not work?:
```python
    @staticmethod
    def _replace_escape(match):
        # Convert to integer using base8 and then convert to character
        return chr(int(match.group()[1:], 8))

    def _replace_octal_escapes(self, value):
        return self.MTAB_FIELD_OCTAL_ESCAPE_RE.sub(_replace_escape, value)
```

Reply:
It does work, but the staticmethod function must still be called with `self._replace_escape`. I've pushed up a new revision that splits the functions like this, as well as adding unit tests